src/Forms/TextInput.tsx

```diff
@@ -0,0 +1,105 @@
+import React, {
+  ReactElement,
+  useContext,
+  ChangeEventHandler,
+  FunctionComponent,
+} from 'react';
+import styled, { ThemeContext } from 'styled-components';
+import { BaseTheme } from 'Theme';
+import ValidationErrorMessage from './ValidationErrorMessage';
+
+/** Represents the possible values for TextInput's label positioning */
+enum POSITION {
+  TOP = 'top',
+  LEFT = 'left',
+}
+
+export interface TextInputProps {
+  /** Function to call on change event */
+  onChange: ChangeEventHandler;
+  /** The id of the text input field */
+  id: string;
+  /** The name of the text input field */
+  name: string;
+  /** The placeholder value of the input field */
+  placeholder?: string;
+  /** Specifies the type of input */
+  type?: string;
+  /** The current value in the text input field */
+  value: string;
+  /** Specifies the size of the font for the text in the input field */
+  fontSize?: string;
+  /** The body of the error message, if applicable */
+  errorMessage?: string;
+  /** Specifies the label text */
+  label: string;
+  /** Allows you to pass in a label position property from the POSITION enum */
+  labelPosition?: POSITION;
+  /** If true, label will be visible */
+  isLabelVisible?: boolean;
+  /** If true, text input field will be disabled */
+  disabled?: boolean;
+}
+
+const StyledTextInput = styled.input<TextInputProps>`
```

Comment:
One irregularity I've noticed is that inputs with `labelPostiion=POSITION.LEFT` render at a partial width, while `labelPostiion=POSITION.TOP` renders at full width. We could fix this by setting `width: 100%;` on this input component, and then allowing the parent container to set the size as needed.

![2020-03-10_105632](https://user-images.githubusercontent.com/6124793/76325587-e0ea1680-62bd-11ea-814a-a788506d315f.png)

![2020-03-10_105628](https://user-images.githubusercontent.com/6124793/76325588-e0ea1680-62bd-11ea-97e6-05f9cc22fae4.png)

Reply:
Updated width for the input component through 29b5864.